Source/multi.cpp

```diff
@@ -129,7 +129,11 @@ byte *ReceivePacket(TBuffer *pBuf, byte *body, size_t *size)
 void NetReceivePlayerData(TPkt *pkt)
 {
 	const Player &myPlayer = *MyPlayer;
-	const Point target = myPlayer.GetTargetPosition();
+	Point target = myPlayer.GetTargetPosition();
+	// Don't send desired target position when we will change our position soon.
+	// This prevents a desync where the remote client starts a walking to the old target position when the teleport is finished but the the new position isn't received yet.
+	if (myPlayer._pmode == PM_SPELL && IsAnyOf(myPlayer.executedSpell.spellId, SpellID::Teleport, SpellID::Phasing, SpellID::Warp))
+		target = {};

```

Comment:
Using `(0, 0)` as the "null" value seems a bit dangerous, consider using a bool or `optional`.

Reply:
We're writing `target.x` and `target.y` to the packet header so we'd either need a sentinel value like this one or add an additional byte to the packet headers. I believe the assumption is that `(0, 0)` is always at the edge of the dungeon and therefore unreachable by the player. But maybe `Point { 0xFF, 0xFF }` would be better, since that's basically guaranteed to be out of bounds of the dungeon.

Reply:
maybe that should be a general change then, there are lots of other cases where 0,0 is used as the null position.

Reply:
maybe that should be a general change then, there are lots of other cases where 0,0 is used as the null position.